### ISSUE TITLE
Avoid using hdf5-1.14.3

### ISSUE THREAD

**Issue by penguian** [ACCESS-NRI/spack-packages#119]
See https://github.com/ACCESS-NRI/ACCESS-ESM1.5/issues/6 
The following change in `packages/mom5/package.py` results in a successful ACCESS-ESM1.5 `pre-industrial` run:
```
[pcl851@gadi-login-09 spack-packages]$ git diff
diff --git a/packages/mom5/package.py b/packages/mom5/package.py
index a36c149..6309dde 100644
--- a/packages/mom5/package.py
+++ b/packages/mom5/package.py
@@ -45,6 +45,9 @@ class Mom5(MakefilePackage):
         depends_on("libaccessom2~deterministic", when="~deterministic")
     with when("@access-esm1.5"):
         depends_on("oasis3-mct@access-esm1.5")
+        # Avoid segfault in HDF5 1.14.3
+        # https://github.com/HDFGroup/hdf5/issues/4381
+        depends_on("hdf5@:1.14.2,1.14.4:")
 
     phases = ["edit", "build", "install"]
```
All Spack builds for ACCESS-ESM1.5 that depend on `hdf5` should include the equivalent of:
```
     with when("@access-esm1.5"):
         depends_on("hdf5@:1.14.2,1.14.4:")
```
to avoid the bug described in https://github.com/HDFGroup/hdf5/issues/4381 and https://github.com/HDFGroup/hdf5/issues/3831

**Comment by penguian** [comment]
@harshula Is there a way in https://github.com/ACCESS-NRI/spack-config to avoid using `hdf5-1.14.3` for any ACCESS-NRI Spack package?

**Comment by harshula** [comment]
https://github.com/spack/spack/blob/9151fc1653317059097ae51a238cbb7c05197240/var/spack/repos/builtin/packages/hdf5/package.py#L48-L54
```
    # Odd versions are considered experimental releases
    # Even versions are maintenance versions
    version(
        "1.14.3",
        sha256="09cdb287aa7a89148c1638dd20891fdbae08102cf433ef128fd345338aa237c7",
        preferred=True,
    )
```

**Comment by penguian** [comment]
The following patch is meant to fix the problem with `hdf5-1.14.3`:

**Comment by harshula** [comment]
Yes, in `spack-config` there's a `packages.yaml` file where we can pin versions of packages.
